src/services/appointment.service.ts

```diff
@@ -108,10 +108,36 @@ export default class AppointmentService implements OnModuleInit {
     updateDto: AppointmentUpdateDto,
   ): Promise<Appointment> {
     const appointment = await this.searchForId(id);
+
+    if (updateDto.customer_name) {
+      const customer = appointment.customer;
```

Comment:
Wouldn't this be shorter?
```suggestion
      if (updateDto.customer_name) {
        appointment.customer.name = updateDto.customer_name;
        await this.customerRepository.save(appointment.customer);
      }
```

Reply:
yep, you´re right, solved!